libgalois/include/katana/PropertyGraphRetractor.h

```diff
@@ -69,10 +69,30 @@ class KATANA_EXPORT PropertyGraphRetractor {
     return pg_->rdg_.edge_properties();
   }
 
+  /// Return true if type information has been loaded separate from properties.
+  /// Return false otherwise.
+  bool IsEntityTypeIDsOutsideProperties() {
+    return pg_->rdg_.IsEntityTypeIDsOutsideProperties();
+  }
+
+  /// Modifiable access to the underlying type array.
+  /// This is exposed because type id mappings change sometimes.
+  PropertyGraph::EntityTypeIDArray& NodeTypeArray() {
```

Comment:
I would still favor returning a pointer to the NUMAArray buffer as opposed to reference to NUMAArray. That way, at least ownership of memory is known.

Reply:
Why does the reference not make ownership of the memory clear?

Reply:
When I see a C pointer I know I don't own the memory.  Mutable references are frowned on (by clang-tidy and I think our coding standard).  It is easy to copy them without realizing it.  At least for me.

Reply:
Okay. Could I do a pointer to the array and not the data buffer?

Reply:
NO, that's worse because that makes it even easier to do bad things to the NUMAArray. A pointer to buffer hides the fact that it was a NUMAArray internally. and like Emmett said, it conveys the message that you don't own the memory.

Reply:
Well then I would like to vote in favor of a buffer class again.

Reply:
I'm gonna implement it any day now. and Arrow fans will tell me that "Arrow has a Buffer already".